From 11cbcdf7088fa33efd9c03f02566f39d600f63a0 Mon Sep 17 00:00:00 2001
From: leaf corcoran <leafot@gmail.com>
Date: Sun, 22 Sep 2019 13:43:18 -0700
Subject: [PATCH] use macro instead of header defines for Lua 5.2+ support

---
 lua_cjson.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lua_cjson.c b/lua_cjson.c
index 875bdaf..1cb44e7 100644
--- a/lua_cjson.c
+++ b/lua_cjson.c
@@ -102,6 +102,10 @@
 #define json_lightudata_mask(ludata)    (ludata)
 #endif
 
+#if LUA_VERSION_NUM > 501
+#define lua_objlen(L,i) lua_rawlen(L, (i))
+#endif
+
 static const char * const *json_empty_array;
 static const char * const *json_array;
 
